File: js/example/src/application/Google/Pages/GoogleBasePage.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var Common = require("../../../../../common/lib/index");
var Google = require("./SearchResults/WebSearchResults");
var GoogleBasePage = (function (_super) {
    __extends(GoogleBasePage, _super);
    function GoogleBasePage(testConfiguration) {
        _super.call(this, testConfiguration);
        this.RegisterSubElement("Query Box", { name: "q" });
        this.RegisterSubElement("Submit Query", { css: "[name=btnK], [name=btnG]" });
        this.EnsureElementLoaded("Query Box", null, "Google page failed to load");
    }
    GoogleBasePage.prototype.Search = function (value) {
        this.Type(value, "Query Box");
        this.Click("Submit Query");
        return this.New(Google.WebSearchResults);
    };
    return GoogleBasePage;
})(Common.WebPage);
exports.GoogleBasePage = GoogleBasePage;

File: Aliases.Example/Application/Google/Pages/GoogleBasePage.cs
using Aliases.Common;
using Aliases.Common.Configuration;
using Aliases.Common.Shared.Types;
using Aliases.Example.Application.Google.Pages.SearchResults;

namespace Aliases.Example.Application.Google.Pages
{
    public class GoogleBasePage : WebPage
    {
        public Alias QueryBox = new Alias() { Name = "q" };
        public Alias SubmitQuery = new Alias() { Css = "[name=btnK], [name=btnG]" };

        public GoogleBasePage(ITestConfiguration testConfig) : base(testConfig)
        {
            EnsureElementLoaded(QueryBox, null, "Google page failed to load.");
        }
        public WebSearchResults Search(string value)
        {
            Type(QueryBox, value);
            Click(SubmitQuery);

            return New<WebSearchResults>();
        }
    }
}
